images/base-notebook/docker_healthcheck.py

```diff
@@ -2,15 +2,20 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 import json
-import os
+import warnings
 from pathlib import Path
 
 import requests
 
+# Import Jupyter paths ignoring deprecation warning that does not affect this script
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Comment:
I think DeprecationWarning is just a warning (so it doesn't stop execution). I would prefer seeing this warning and not ignoring it.

Reply:
> (so it doesn't stop execution)

That's correct, the warning does not affect the actual functionality of the healthcheck script.

The reason I decided to silence the warning is that it might get confusing for users.
With the warning visible, the output of the healtcheck script looks like this:
```
$ docker exec jupyter /etc/jupyter/docker_healthcheck.py
/etc/jupyter/docker_healthcheck.py:13: DeprecationWarning: Jupyter is migrating its paths to use standard platformdirs
given by the platformdirs library.  To remove this warning and
see the appropriate new directories, set the environment variable
`JUPYTER_PLATFORM_DIRS=1` and then run `jupyter --paths`.
The use of platformdirs will be the default in `jupyter_core` v6
  from jupyter_core import paths
b'{"version": "2.12.2"}'
```

Users will not normally run this script manually, but in case the container is unhealthy, they will see this output from Docker when reporting the state of the healthcheck failure.

Note that the output of the `jupyter --runtime-dir` command does not emit any deprecation warnings because the warning only appears when `jupyter_core.paths` is imported from a non "internal" module ([see source here](https://github.com/jupyter/jupyter_core/blob/9ea909732fda6b7a674a8b7aa8233602e0b6af52/jupyter_core/utils/__init__.py#L76-L90)).

I thought of preserving the same UX for the Docker healthcheck script while directly importing the `jupyter_core.paths` module and not having to externally call the `jupyter --runtime-dir` command as a subprocess.

Let me know if you still would prefer the warning to be emitted.

Reply:
Let's call a subprocess with `jupyter --runtime-dir`, so we don't have to suppress warnings.
This has one more benefit for you - you don't have to deal with `mypy` at all 🙂

Reply:
I just saw this comment now.
Yes I also contemplated that option (calling `jupyter --runtime-dir` as a command) but I thought it is a bit overkill considering it would be a Python -> Shell -> Python chain of processes for a single line of Python code :)

Can you confirm you really prefer to not supress the warning so I proceed with that?

Reply:
I don't thinks it's an overkill, if it makes code look simpler, unlikely to break and doesn't produce warnings.
Yes, I confirm I prefer subprocess in this case 🙂